views/013_check_summary.sql

```diff
@@ -0,0 +1,31 @@
+-- Materialized view for check status summary
+CREATE MATERIALIZED VIEW IF NOT EXISTS check_status_summary AS
+    SELECT
+        check_id,
+        PERCENTILE_DISC(0.99) WITHIN GROUP (ORDER BY duration) as p99,
+        COUNT(*) FILTER ( WHERE status = TRUE) as passed,
+        COUNT(*) FILTER ( WHERE status = FALSE) as failed,
+        MAX(time) as last_check,
+        MAX(time) FILTER (WHERE status = TRUE) as last_pass,
+        MAX(time) FILTER (WHERE status = FALSE) as last_fail
+    FROM
+        check_statuses
+    WHERE
+        time > (NOW() at TIME ZONE 'utc' - Interval '1 hour')
+    GROUP BY check_id;
+
+-- For last transition
+CREATE OR REPLACE FUNCTION update_last_transition_time_for_check()
+RETURNS TRIGGER AS $$
+BEGIN
+    UPDATE checks SET last_transition_time = NOW() WHERE id = NEW.id;
```

Comment:
shouldn't this be NEW.last_transition_time = now() with a before trigger ?

Reply:
yeah, that is much more efficient